jellyfin_kodi/full_sync.py

```diff
@@ -417,37 +417,47 @@ def music(self, library, dialog):
 
-                    # Get all items in the library to process locally
-                    artists_data = server.get_artists(library_id)
-                    artists = artists_data['Items']
-                    num_artists = artists_data['TotalRecordCount']
-                    albums = server.get_library_items(library_id, 'MusicAlbum')['Items']
-                    songs = server.get_library_items(library_id, 'Audio')['Items']
-
-                    for index, artist in enumerate(artists):
-                        artist_name = artist.get('Name')
-
-                        # Update percentage dialog
-                        percent = int((float(index) / float(num_artists)) * 100)
-                        dialog.update(percent, heading="%s: %s" % (translate('addon_name'), library['Name']), message=artist_name)
-
-                        # Add artist to database
-                        obj.artist(artist)
-
-                        # Get all albums for each artist
-                        artist_albums = [album for album in albums if artist_name in album.get('Artists')]
-                        for album in artist_albums:
-                            # Add album to database
-                            obj.album(album)
-                            album_id = album.get('Id')
-                            # Get all songs in each album
-                            album_songs = [song for song in songs if album_id == song.get('AlbumId')]
-                            for song in album_songs:
-                                dialog.update(percent,
-                                              message="%s/%s/%s" % (artist_name, album['Name'][:7], song['Name'][:7]))
-                                # Add song to database
-                                obj.song(song)
-#
+                    total_items = server.get_item_count(library_id, 'MusicArtist,MusicAlbum,Audio')
+                    count = 0
+
+                    '''
+                    Music database syncing.  Artists must be in the database
+                    before albums, albums before songs.  Pulls batches of items
+                    in sizes of setting "Paging - Max items".  'artists',
+                    'albums', and 'songs' are generators containing a dict of
+                    api responses
+                    '''
+                    artists = server.get_artists(library_id)
+                    for batch in artists:
+                        for item in batch['Items']:
+                            LOG.debug('Artist: {}'.format(item.get('Name')))
+                            percent = int((float(count) / float(total_items)) * 100)
+                            dialog.update(percent, message='Artist: {}'.format(item.get('Name')))
+                            obj.artist(item)
+                            count += 1
+                            # Delete item once it's been processed for memory management
+                            del item
```

Comment:
Does this actually delete anything from the dict to save memory? doesn't it just get replaced with the next item when the loop jumps back to the top?

Reply:
My impression was that if we went back and looped through `artists` a second time, all of these items would still be there and accessible again, in which case deleting them should help free memory up.  I haven't explicitly tested that though.

Reply:
Test shows that it did not in fact do anything to del the item there.